routes/api/index.js

```diff
@@ -73,6 +73,32 @@ router.use(
   })
 );
 
+router.use(
+  '/search/similar',
+  routeFactory.defaultSearchRoute({
+    authProvider: authProvider,
+    namedOptions: 'similar', // default: 'all'
+    // options: {
+    //   Add JSON search options here.
+    //   default: none
+    //   WARNING: This will override the saved search options referenced by 'namedOptions' above.
+    //   Example for making result labels using name property of person sample-data
+    //   'extract-document-data': {
+    //     'extract-path': ['/name']
+    //   }
+    // },
+    idConverter: idConverter, // default: encodeURIComponent(result.uri)
+    makeLabel: result => {
+      // default: none
+      return (
+        result.extracted &&
+        result.extracted.content[0] &&
+        result.extracted.content[0].name
+      );
+    }
+  })
+);
+
```

Comment:
Sorry for taking so long, but after rethinking about this, I am thinking we should not need a separate /search/similar endpoint. Instead, similar should just be 'yet another constraint' in the existing all.xml. Could you revisit this PR (you will have to open a new one unfortunately), and move that custom constraint into all.xml?

https://github.com/marklogic-community/grove-ml-gradle/pull/3/files